examples/spawn/main.cpp

```diff
@@ -0,0 +1,250 @@
+    }
+
+    void print_links() {
+        //std::lock_guard<decltype(print_mt)> mt(print_mt);
+        std::cout << "bot3 linked with:\n";
+        auto addrs = all_view_address();
+        for (const auto& a : addrs) {
+            std::cout << '\t' << a << std::endl;
+        }
+    }
+};
+
+class worker_t final : public actor_zeta::basic_async_actor {
+    auto get_info() -> std::string {
+        std::string s;
+        std::stringstream ss(s);
+        ss << "class:bot1 type:" << this->type() << "(" << address().get() << ")";
+        return ss.str();
+    }
+
+public:
+    explicit worker_t(actor_zeta::supervisor_abstract* ptr /*, int a, double b*/)
+        : actor_zeta::basic_async_actor(ptr, "bot1") {
+        count++;
+        std::cout << get_info() << " created" << std::endl;
+        add_handler("spawn_actor_in_bot", &worker_t::spawn_worker);
+        add_handler("spawn_broadcast", &worker_t::spawn_broadcast);
+        add_handler("print_links", &worker_t::print_links);
+    }
+
+    auto spawn_worker() -> void {
+        auto addr = address_book("manager-2");
+        actor_zeta::spawn_actor<worker_t2>(addr, address());
+        actor_zeta::spawn_actor<worker_t3>(addr, address());
+    }
+
+    void spawn_broadcast(actor_zeta::address_t addr) {
+        std::cout << get_info() << " got actor: " << addr << " " << addr.type() << " " << addr.get() << std::endl;
+        if (addr.type() == "bot2") {
+            auto self_addr = address();
+            actor_zeta::link(self_addr, addr);
+        }
+    }
+
+    void print_links() {
+        //std::lock_guard<decltype(print_mt)> mt(print_mt);
+        std::cout << get_info() << " linked with:\n";
+        auto addrs = all_view_address();
+        for (const auto& a : addrs) {
+            std::cout << '\t' << a << std::endl;
+        }
+    }
+};
+
+class supervisor_lite final : public actor_zeta::supervisor_abstract {
+public:
+    explicit supervisor_lite(std::string name)
+        : supervisor_abstract(std::move(name))
+        , e_(new executor_t<work_sharing>(1, max_queue),
+             thread_pool_deleter)
+        , system_{"add_link",
+                  "sync_contacts",
+                  "remove_link",
+                  "spawn_actor", "delegate", "spawn_broadcast"} {
+        e_->start();
+        add_handler("spawn_broadcast", &supervisor_lite::spawn_broadcast);
+    }
+
+    auto executor_impl() noexcept -> actor_zeta::abstract_executor* final { return e_.get(); }
+
+    auto add_actor_impl(actor_zeta::actor t) -> void final {
+        std::cout << "sup: added actor:" << t.address() << " " << t.type() << " " << t->address().get() << std::endl;
+        spawn_broadcast(address(), t->address());
+        actors_.emplace_back(std::move(t));
+    }
+
+    auto add_supervisor_impl(actor_zeta::supervisor t) -> void final {
+        supervisor_.emplace_back(std::move(t));
+    }
+
+    auto spawn_broadcast(actor_zeta::address_t sender, actor_zeta::address_t addr) -> void {
```

Comment:
use the built-in brotcast function

Reply:
look here please  :  https://github.com/cyberduckninja/actor-zeta/blob/develop/header/actor-zeta/base/communication_module.hpp#L83